packages/mdc-select/foundation.ts

```diff
@@ -111,9 +121,11 @@ export class MDCSelectFoundation extends MDCFoundation<MDCSelectAdapter> {
     const previouslySelectedIndex = this.selectedIndex_;
     this.selectedIndex_ = index;
 
-    this.adapter_.setSelectedText(
-        this.selectedIndex_ === numbers.UNSET_INDEX ?
-            '' : this.adapter_.getMenuItemTextAtIndex(this.selectedIndex_).trim());
+    if (this.selectedIndex_ === numbers.UNSET_INDEX || !this.adapter_.getMenuItemTextAtIndex(this.selectedIndex_)) {
```

Comment:
When'll this return falsy value?

Reply:
If `index` is out of range. Alternatively we could add an adapter method `getNumberOfMenuItems` to make sure that it is in range?

Reply:
Adapter to get menu count sounds good. Foundation can throw error when index is not valid.